### ISSUE TITLE
`platformAutomerge` doesn't work on GHES

### ISSUE THREAD

**Issue by Shegox** [renovatebot/renovate#15921]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.66.5

### Please select which platform you are using if self-hosting.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub Enterprise Server 3.3.8

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I enabled [`platformAutomerge`](https://docs.renovatebot.com/configuration-options/#platformautomerge), which got enabled fully supported in GHES version >=3.3.0 and it was ignored.

Renovate config: 
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "platformAutomerge": true,
  "ignoreTests": true,
  "packageRules": [
    {
      "matchPackageNames": ["renovate/renovate"],
      "automerge": true
    }
  ]
}
```

Looking at the code it seems that it is currently completely disabled for GHES (see #12464 for details):
https://github.com/renovatebot/renovate/blob/381481ace1fae9407a10295520643af4bfd4fca0/lib/modules/platform/github/index.ts#L1326-L1328

I would propose to update the precheck to ensure that if the GHE version >= 3.3 like this:

```ts
  // Only GitHub Enterprise Server >= 3.3.0 supports automerge
  if (
    platformConfig.isGhe &&
    semver.satisfies(platformConfig.gheVersion!, '<3.3.0')
  ) {
    return;
  }

  if (!platformOptions?.usePlatformAutomerge) {
    return;
  }
```

If that is desired I can open a PR and get it implemented/tested?

### Relevant debug logs

<details><summary>Logs</summary>

```
2022-06-07T06:57:20.478497691Z DEBUG: Setting current branch to main (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.529376345Z DEBUG: latest commit (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.529406304Z        "branchName": "main",
2022-06-07T06:57:20.529409157Z        "latestCommitDate": "2022-06-07T08:46:20+02:00"
2022-06-07T06:57:20.548816000Z DEBUG: getBranchPr(renovate/renovate-renovate-30.x) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.548947499Z DEBUG: findPr(renovate/renovate-renovate-30.x, undefined, open) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.549095444Z DEBUG: findPr(renovate/renovate-renovate-30.x, undefined, closed) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.549188967Z DEBUG: Found PR #10 (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.549305686Z DEBUG: branchExists=false (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.549411685Z DEBUG: dependencyDashboardCheck=unlimit (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.549621172Z DEBUG: recreateClosed is false (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.549780575Z DEBUG: findPr(renovate/renovate-renovate-30.x, Update dependency renovate/renovate to v30.4.1, !open) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.549838480Z DEBUG: prAlreadyExisted=false (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.549927289Z DEBUG: Checking schedule(null, null) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.549978899Z DEBUG: No schedule defined (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.550150043Z DEBUG: Branch needs creating (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.550225861Z DEBUG: Using reuseExistingBranch: false (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.550319762Z DEBUG: manager.getUpdatedPackageFiles() reuseExistinbranch=false (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.566929941Z DEBUG: Starting search at index 5 (repository=dummy-testing/renovate-controller, packageFile=renovate.Dockerfile, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.566952298Z        "depName": "renovate/renovate"
2022-06-07T06:57:20.567186707Z DEBUG: Found match at index 5 (repository=dummy-testing/renovate-controller, packageFile=renovate.Dockerfile, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.567194857Z        "depName": "renovate/renovate"
2022-06-07T06:57:20.568083458Z DEBUG: Contents updated (repository=dummy-testing/renovate-controller, packageFile=renovate.Dockerfile, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.568093221Z        "depName": "renovate/renovate"
2022-06-07T06:57:20.568426465Z DEBUG: Updated 1 package files (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.568559590Z DEBUG: Getting updated lock files (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.569072679Z DEBUG: Writing package.json files (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.569081409Z        "packageFiles": ["package.json"]
2022-06-07T06:57:20.569252389Z DEBUG: Writing package-lock.json (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.611054439Z DEBUG: Writing any updated package files (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.613908610Z DEBUG: No updated lock files in branch (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.614158346Z DEBUG: 1 file(s) to commit (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.614744284Z DEBUG: Preparing files for committing to branch renovate/renovate-renovate-30.x (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.817256670Z DEBUG: git commit (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:20.817272849Z        "deletedFiles": [],
2022-06-07T06:57:20.817275556Z        "ignoredFiles": [],
2022-06-07T06:57:20.817277713Z        "result": {
2022-06-07T06:57:20.817279952Z          "author": null,
2022-06-07T06:57:20.817282029Z          "branch": "renovate/renovate-renovate-30.x",
2022-06-07T06:57:20.817284093Z          "commit": "dc432af7500ece97b38692c561894841275fc066",
2022-06-07T06:57:20.817286009Z          "root": false,
2022-06-07T06:57:20.817288182Z          "summary": {"changes": 1, "insertions": 1, "deletions": 1}
2022-06-07T06:57:20.817290461Z        }
2022-06-07T06:57:20.837395612Z DEBUG: Pushing branch renovate/renovate-renovate-30.x (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.242715182Z DEBUG: git push (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.242736057Z        "result": {
2022-06-07T06:57:24.242738712Z          "pushed": [
2022-06-07T06:57:24.242740987Z            {
2022-06-07T06:57:24.242743412Z              "deleted": false,
2022-06-07T06:57:24.242745579Z              "tag": false,
2022-06-07T06:57:24.242747521Z              "branch": true,
2022-06-07T06:57:24.242749468Z              "new": true,
2022-06-07T06:57:24.242751557Z              "alreadyUpdated": false,
2022-06-07T06:57:24.242754007Z              "local": "refs/heads/renovate/renovate-renovate-30.x",
2022-06-07T06:57:24.242755837Z              "remote": "refs/heads/renovate/renovate-renovate-30.x"
2022-06-07T06:57:24.242757684Z            }
2022-06-07T06:57:24.242759852Z          ],
2022-06-07T06:57:24.242762586Z          "ref": {"local": "refs/remotes/origin/renovate/renovate-renovate-30.x"},
2022-06-07T06:57:24.242764653Z          "remoteMessages": {
2022-06-07T06:57:24.242766750Z            "all": [
2022-06-07T06:57:24.242768751Z              "Create a pull request for 'renovate/renovate-renovate-30.x' on GitHub by visiting:",
2022-06-07T06:57:24.242770967Z              "https://<redacted>/dummy-testing/renovate-controller/pull/new/renovate/renovate-renovate-30.x"
2022-06-07T06:57:24.242773218Z            ],
2022-06-07T06:57:24.242775455Z            "pullRequestUrl": "https://<redacted>/dummy-testing/renovate-controller/pull/new/renovate/renovate-renovate-30.x"
2022-06-07T06:57:24.242777626Z          }
2022-06-07T06:57:24.242779600Z        }
2022-06-07T06:57:24.242984626Z  INFO: Branch created (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.242999782Z        "commitSha": "dc432af7500ece97b38692c561894841275fc066"
2022-06-07T06:57:24.243078740Z DEBUG: Checking if we can automerge branch (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.243150747Z DEBUG: mergeStatus=no automerge (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.243210375Z DEBUG: Ensuring PR (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.243267111Z DEBUG: There are 0 errors and 0 warnings (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.243412615Z DEBUG: getBranchPr(renovate/renovate-renovate-30.x) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.243471374Z DEBUG: findPr(renovate/renovate-renovate-30.x, undefined, open) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.243546961Z DEBUG: findPr(renovate/renovate-renovate-30.x, undefined, closed) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.243603633Z DEBUG: Found PR #10 (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.252196431Z DEBUG: resolveBranchStatus(branchName=renovate/renovate-renovate-30.x, ignoreTests=true) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.252313943Z DEBUG: Ignore tests. Return green (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.255092970Z DEBUG: Creating PR (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.255110196Z        "prTitle": "Update dependency renovate/renovate to v30.4.1"
2022-06-07T06:57:24.255570641Z DEBUG: Creating PR (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:24.255582596Z        "title": "Update dependency renovate/renovate to v30.4.1",
2022-06-07T06:57:24.255585506Z        "head": "dummy-testing:renovate/renovate-renovate-30.x",
2022-06-07T06:57:24.255587829Z        "base": "main",
2022-06-07T06:57:24.255589749Z        "draft": false
2022-06-07T06:57:25.087759029Z DEBUG: PR created (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.087786412Z        "pr": 15,
2022-06-07T06:57:25.087790827Z        "draft": false
2022-06-07T06:57:25.088060011Z DEBUG: Adding labels '' to #15 (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.091960004Z  INFO: PR created (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.091982473Z        "pr": 15,
2022-06-07T06:57:25.091986461Z        "prTitle": "Update dependency renovate/renovate to v30.4.1"
2022-06-07T06:57:25.092081230Z DEBUG: resolveBranchStatus(branchName=renovate/renovate-renovate-30.x, ignoreTests=true) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.092185772Z DEBUG: Ignore tests. Return green (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.092306952Z DEBUG: Skipping assignees and reviewers as automerge=true (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.092408473Z DEBUG: Created Pull Request #15 (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.092509487Z DEBUG: PR is configured for automerge (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.093038183Z DEBUG: resolveBranchStatus(branchName=renovate/renovate-renovate-30.x, ignoreTests=true) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.093150768Z DEBUG: Ignore tests. Return green (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.093245781Z DEBUG: Automerging #15 with strategy auto (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.093760260Z DEBUG: mergePr(15, renovate/renovate-renovate-30.x) (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.093933875Z DEBUG: Branch protection: Attempting to merge PR when PR reviews are enabled (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.093941228Z        "prNo": 15
2022-06-07T06:57:25.230771179Z DEBUG: Branch protection: Cannot automerge PR until there is an approving review (repository=dummy-testing/renovate-controller, branch=renovate/renovate-renovate-30.x)
2022-06-07T06:57:25.230805803Z        "prNo": 15
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
You need to check not just when platform automerge was added to GHES but also when the necessary GraphQL mutations were added. GraphQL often trails web UI.

**Comment by Shegox** [comment]
Good point 👍 

The docs state that the GraphQL mutation is available from 3.1 onwards, I would [opt for 3.3 though](https://docs.github.com/en/enterprise-server@3.3/graphql/reference/mutations#enablepullrequestautomerge), because in 3.1 and 3.2 the REST API doesn't output if auto_merge is allowed for a repository or not.

I have both an 3.2 and 3.3 GitHub Enterprise Server I can check the changes against.
